src/api/c++/z3++.h

```diff
@@ -3615,13 +3757,17 @@ namespace z3 {
             return c ? c : (Z3_context)s->ctx();
         }
 
-        struct scoped_cb {
+        struct scoped_cb final {
             user_propagator_base* p;
             scoped_cb(void* _p, Z3_solver_callback cb):p(static_cast<user_propagator_base*>(_p)) {
                 p->cb = cb;
             }
-            ~scoped_cb() { 
-                p->cb = nullptr; 
+            scoped_cb(scoped_cb const &) = default;
```

Comment:
this is a private class. doesn't need any change.

Reply:
Makes sense. Changing.